src/tools/db-upgrade/DbUpgraderTool.cpp

```diff
@@ -164,15 +174,96 @@ int main(int argc, char* argv[]) {
   CHECK_NOTNULL(schemaMan);
   CHECK_NOTNULL(indexMan);
 
-  if (FLAGS_upgrade_version != 1 && FLAGS_upgrade_version != 2) {
-    LOG(ERROR) << "Flag upgrade_version : " << FLAGS_upgrade_version
-               << " illegal, upgrade_version can only be 1 or 2";
+  if (FLAGS_raw_data_version != 1 && FLAGS_raw_data_version != 2) {
+    LOG(ERROR) << "Flag raw_data_version : " << FLAGS_raw_data_version
+               << " illegal, raw_data_version can only be 1 or 2";
     return EXIT_FAILURE;
   }
-  LOG(INFO) << "Prepare phase end";
+
+  // one space, check partId
+  std::vector<std::string> parts;
+  std::unordered_set<PartitionID> partsAll;
+  folly::split(",", FLAGS_partIds, parts, true);
+  for (auto& part : parts) {
+    PartitionID partId;
+    try {
+      partId = folly::to<PartitionID>(part);
+    } catch (const std::exception& ex) {
+      LOG(ERROR) << "Cannot convert part id " << part;
+      return EXIT_FAILURE;
+    }
+    partsAll.emplace(partId);
+  }
+
+  for (auto& path : srcPaths) {
+    auto dataPath = nebula::fs::FileUtils::joinPath(path, "nebula");
+    if (!nebula::fs::FileUtils::exist(dataPath)) {
+      LOG(ERROR) << "Source data path " << dataPath << " not exists!";
+      return EXIT_FAILURE;
+    }
+    auto subDirs = nebula::fs::FileUtils::listAllDirsInDir(dataPath.c_str());
+    if (subDirs.size() != 1) {
+      LOG(ERROR) << "Only handle one space!";
+      return EXIT_FAILURE;
+    }
+    auto space = subDirs[0];
+    GraphSpaceID spaceId;
+    try {
+      spaceId = folly::to<GraphSpaceID>(space);
+    } catch (const std::exception& ex) {
+      LOG(ERROR) << "Cannot convert space id " << space;
+      return EXIT_FAILURE;
+    }
+
+    auto sRet = schemaMan->toGraphSpaceName(spaceId);
```

Comment:
spaceName is useless?

Reply:
Here only check whether the space exists or not